File: 009_readability/readability.py
from cs50 import get_string

# Get the text.
text = get_string("Text: ")

# Set up the counters.
letter_count = 0
space_count = 1
punct_count = 0

for i in range(len(text)):
    char = text[i]

    # Check if the character is a letter.
    if char.isalpha():
        letter_count += 1
    # Check if the character is a space.
    elif char.isspace():
        space_count += 1
    # Check if the character is a "!", "?", or "."
    elif char == "!" or char == "?" or char == ".":
        punct_count += 1

# Calculate the grade.
l = letter_count / space_count * 100
s = punct_count / space_count * 100
grade = round(0.0588 * l - 0.296 * s - 15.8)

# Print the grade.
if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: 009_readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    // Get the text
    char *text = get_string("Text: ");

    // Get first character in string
    int i = 0;
    char c = text[i];

    // Set up counters
    float letter_count = 0.0;
    float space_count = 1.0;
    float punct_count = 0.0;

    while (c != 0)
    {
        // Check if character is a letter
        if (isalpha(c))
        {
            letter_count += 1;
        }
        // Check if character is a space
        else if (isspace(c))
        {
            space_count += 1;
        }
        // Check if character is a "!", "?", or "."
        else if (c == 33 || c == 63 || c == 46)
        {
            punct_count += 1;
        }

        i += 1;
        c = text[i];
    }

    // Calculate the grade
    double l = letter_count / space_count * 100;
    double s = punct_count / space_count * 100;
    int grade = round(0.0588 * l - 0.296 * s - 15.8);

    // Return grade
    if (grade > 16)
    {
        printf("Grade 16+\n", grade);
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n", grade);
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}